decoder.go

```diff
@@ -199,6 +194,27 @@ func (d *decoder) setFieldByType(current reflect.Value, namespace []byte, idx in
 			}
 		}
 	}
+	{
+		v := v // deliberately shadow v as to not modify
+		t := v.Type()
+		if t.Kind() != reflect.Ptr && v.CanAddr() {
+			v = v.Addr()
+		}
+		if um, ok := v.Interface().(Unmarshaler); ok {
+			// if receiver is a nil pointer, set before calling function.
+			if t.Kind() == reflect.Ptr && v.IsNil() {
+				t = t.Elem()
+				v.Set(reflect.New(t))
+				um = v.Interface().(Unmarshaler)
+			}
```

Comment:
Is there any way to avoid doing this/duplicating the `Ptr` logic of creating the new element?

below this same logic exists for `Ptr` already https://github.com/go-playground/form/pull/63/files#diff-2a967f84f1af0d52111b8dbfbeadc44bfeb1092dacebb2bc0648a0e57284c04bR226-R230.

It would be great if it's possible to keep the flow of the program using that same logic.

Reply:
I'll have a look at this and see if anything can be done